.github/workflows/main.yml

```diff
@@ -11,6 +11,21 @@ jobs:
     - uses: actions/setup-node@v1
       with:
         node-version: '10'
+
     - run: npm install
-    - run: npm test
-    - run: npm run validate
+
+    - name: test scripts
+      run: npm test
+
+    - name: validate registries
+      run: npm run validate
+
+    - name: canonicalize registries
+      if: github.ref != 'refs/heads/master'
+      run: |
+          npm run canonicalize
+          git config --global user.name 'ISDCF Registries Canonicalizer'
+          git config --global user.email 'admin@isdcf.com'
+          git add src/main/data
+          git commit -m "Registries have been canonicalized"
+          git push
```

Comment:
This is great! So I guess if there are no changes, then `git commit ...` does nothing, and `git push` does nothing?

Reply:
Yes. We need to try on a fork to see if GitHub Actions is triggered there. If not, we will need to generate PR upon merge into master.

Reply:
Here's the proposal:

- CI fails if registries are not canonicalized when either (a) committing to master or (b) creating a PR from a fork
- CI automatically creates a commit with canonicalized registries on PR from this registry

Reply:
I'm surprised that (b) isn't possible, but I think we can live with these for now. Do you feel like they're possible to fix with more tweaks? Just wondering if I should play around, or if you've confirmed they're no-go.

Reply:
AFAIK a PR from a fork does not create a branch in the local repo, so there is nothing to commit against. It is possible for the forked repo to enable GitHub Actions, but that does not help us.

Let me know if you find otherwise.